src/connectors/search-box/connectSearchBox.js

```diff
@@ -139,6 +133,19 @@ export default function connectSearchBox(renderFn, unmountFn = noop) {
         return state.setQueryParameter('query', undefined);
       },
 
+      getWidgetRenderState(renderState, { helper, searchMetadata }) {
+        return {
+          ...renderState,
+          searchBox: {
+            query: helper.state.query || '',
+            refine: this._refine,
+            clear: this._cachedClear,
+            widgetParams,
+            isSearchStalled: searchMetadata.isSearchStalled,
+          },
+        };
+      },
```

Comment:
I didn't want to rush that decision now because it might be too naive. There are probably things that we'll miss. E.g., inability for a widget to be responsible of multiple UI states—that could be the case if `infiniteHits` starts controlling another UI state than `page`. Besides, that'd be a breaking change.

Reply:
The point of the discussion was that if we do the merge ourself it would not be a breaking change and it should not have any impact on the widgets that return a full UI State in all cases.

Reply:
To my first point, I don't think this is the right time to do this, and we'll need proper testing to decide if we want to do this at some point.

Reply:
It feels like a missed opportunity here but ok.